libs/components/src/lib/popup/popup.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { elementUpdated, fixture, getControlElement } from '@vivid-nx/shared';
+import { Corner, Position } from '../enums';
+import type { Button } from '../button/button';
+import { Popup } from './popup';
+import '.';
+
+const COMPONENT_TAG = 'vwc-popup';
+
+global.ResizeObserver = jest.fn().mockImplementation(() => ({
+	observe: jest.fn(),
+	unobserve: jest.fn(),
+	disconnect: jest.fn(),
+}));
+
+
+describe('vwc-popup', () => {
+	let element: Popup;
+
+	beforeEach(async () => {
+		element = await fixture(`<${COMPONENT_TAG}></${COMPONENT_TAG}>`) as Popup;
+	});
+
+	describe('basic', () => {
+		it('initializes as a vwc-popup', async () => {
+			expect(element).toBeInstanceOf(Popup);
+			expect(element.open).toBeFalsy();
+			expect(element.arrow).toBeFalsy();
+			expect(element.dismissible).toBeFalsy();
+			expect(element.anchor).toEqual('');
+			expect(element.corner).toEqual(Corner.Left);
+			expect(element.strategy).toEqual(Position.Fixed);
+		});
+	});
+
+	describe('show', () => {
+		it('should set "open" to true', async () => {
+			await setPopupAndAnchor();
+			element.anchor = 'anchor';
+			await elementUpdated(element);
+
+			element.show();
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+	});
+
+	describe('hide', () => {
+		it('should set "open" to false', async () => {
+			element.open = true;
+			element.hide();
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(false);
+		});
+	});
+
+	describe('anchor', () => {
+		it('should not set popup open if anchor element does not exist', async () => {
+			element.anchor = 'anchor';
+			await elementUpdated(element);
+
+			element.show();
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(false);
+		});
+	});
+
+	describe('render classes', () => {
+		it('should render arrow class', async () => {
+			element.arrow = true;
+			await elementUpdated(element);
+			expect(element.shadowRoot?.querySelector('.arrow')).not.toBeNull();
+		});
+
+		it('should render dismiss class', async () => {
+			element.dismissible = true;
+			await elementUpdated(element);
+			expect(element.shadowRoot?.querySelector('.dismissible')).not.toBeNull();
+		});
+	});
```

Comment:
These are two different API's.
```suggestion
	describe('render arrow', () => {
		it('should render an arrow icon if arrow is true', async () => {
			// TODO
		});
		
		it('should not render an arrow icon if arrow is false', async () => {
			// TODO
		});
		
		it('should remove the arrow class on the container if arrow is false', () => {
		        const controlElement = element.shadowRoot?.querySelector('.control');
			expect(controlElement.classList.contains('.arrow')).toEqual(false);
		});
		
		it('should set the arrow class on the container if arrow is true', () => {
		        element.arrow = true;
			await elementUpdated(element);
			const controlElement = element.shadowRoot?.querySelector('.control');
			expect(controlElement.classList.contains('.arrow')).toEqual(true);
		});
	});
```
This suggestion only covers the idea for `arrow`. Do the same for dismissible.

Reply:
@YonatanKra It won't work, the arrow and dismiss classes are not inside the control.
Also the "should remove the arrow class on the container if arrow is false" check is not necessary cause "initializes as a vwc-popup" test already checks it.

Reply:
That's exactly why we should add the correct test to document where we should expect the class.
BTW - is this class exposed as API? I think what's more important is that we have an icon/button, no?